katerina.oblakevich/task_5/src/main/java/com/hillel/BankApp.java

```diff
@@ -0,0 +1,132 @@
+import com.hillel.model.BankAccount;
+import com.hillel.model.ReqStatus;
+import com.hillel.model.User;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+@WebServlet(name = "BankServlet", urlPatterns = {"/accounts"})
+public class BankApp extends HttpServlet {
+    private List<BankAccount> bankAccounts = fillAccountList();
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        Gson gson = new Gson();
+        String parameter = req.getParameter("acc_id");
+
+        if (parameter != null) {
+            int id = Integer.parseInt(parameter);
+            for (BankAccount b : bankAccounts) {
+                if (b.getId() == id) {
+                    resp.getWriter().println(gson.toJson(b));
+                    return;
+                }
+            }
+            resp.setStatus(404);
+            resp.getOutputStream().print(gson.toJson(new ReqStatus(false, "Account not found")));
+            resp.getOutputStream().flush();
+        } else
+            resp.getWriter().println(gson.toJson(bankAccounts));
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        Gson gson = new Gson();
+        ReqStatus status;
+
+        String parId = req.getParameter("acc_id");
+        String parAmount = req.getParameter("new_amount");
+        if ((parId != null) && (parAmount != null)) {
+            int id = Integer.parseInt(parId);
+            long newAmount = Long.parseLong(parAmount);
+            for (BankAccount b : bankAccounts) {
+                if (b.getId() == id) {
+                    b.setAmount(newAmount);
+                    status = new ReqStatus(true, "Set new amount successful");
+                    resp.getOutputStream().print(gson.toJson(status));
+                    resp.getOutputStream().flush();
+                    return;
+                }
+            }
+            resp.setStatus(404);
+            resp.getOutputStream().print(gson.toJson(new ReqStatus(false, "No book with this id")));
+            resp.getOutputStream().flush();
+        }
+    }
+
+    @Override
+    protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json");
+        Gson gson = new Gson();
+        ReqStatus status;
+
+        try {
+            StringBuilder sb = new StringBuilder();
+            String s;
+            while ((s = req.getReader().readLine()) != null)
+                sb.append(s);
+
+            BankAccount bankAccount = gson.fromJson(sb.toString(), BankAccount.class);
```

Comment:
It was old example. Now we can use req.getReader instead of sb.toString() 
StringBuilder code can be deleted. It's not mistake but recommended